File: LeetCode/0461-hamming-distance/solution.java
class Solution {
    public int hammingDistance(int x, int y) {
        int aux = x^y;
        int answer = 0;
        if(aux<0){
            aux = aux^(-1);
            for(int i = 0; i<32;i++){
                if(aux%2 == 1){
                    answer++;
                }
                aux = aux>>1;
            }
            answer = 32 - answer;
        }else{
            for(int i = 0; i<32;i++){
                if(aux%2 == 1){
                    answer++;
                }
                aux = aux>>1;
            }   
        }
        return answer;
    }
}

File: LeetCode/0461-hamming-distance/solution.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int aux = 0;
        for(int i = 0; i<32;i++){
            if(x%2 != y%2){
                aux++;
            }
            x>>=1;
            y>>=1;
        }
        return aux;
    }
};
